src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -1608,23 +1692,32 @@ JCExpression term3Rest(JCExpression t, List<JCExpression> typeArgs) {
             } else if (token.kind == DOT) {
                 nextToken();
                 typeArgs = typeArgumentsOpt(EXPR);
-                if (token.kind == SUPER && (mode & EXPR) != 0) {
+                if (token.kind == SUPER && isMode(EXPR)) {
                     selectExprMode();
                     t = to(F.at(pos1).Select(t, names._super));
                     nextToken();
                     t = arguments(typeArgs, t);
                     typeArgs = null;
-                } else if (token.kind == NEW && (mode & EXPR) != 0) {
+                } else if (token.kind == NEW && isMode(EXPR)) {
                     if (typeArgs != null) return illegal();
                     selectExprMode();
                     int pos2 = token.pos;
                     nextToken();
                     if (token.kind == LT) typeArgs = typeArguments(false);
                     t = innerCreator(pos2, typeArgs, t);
                     typeArgs = null;
-                } else {
+                } else if (token.kind == TokenKind.STRINGFRAGMENT ||
+                           token.kind == TokenKind.STRINGLITERAL) {
+                    if (typeArgs != null) {
+                        return illegal();
+                    }
+                    t = stringTemplate(t);
+                    typeArgs = null;
```

Comment:
`typeArgs = null;` unnecessary?

Reply:
Will remove the section for now and make a note elsewhere.